packages/warehouses/src/warehouseClients/AthenaWarehouseClient.ts

```diff
@@ -0,0 +1,320 @@
+    INTERVAL_DAY_TIME = 'interval day to second',
+    ARRAY = 'array',
+    MAP = 'map',
+    ROW = 'row',
+    IPADDRESS = 'ipaddress',
+    UUID = 'uuid',
+}
+
+interface TableInfo {
+    database: string;
+    schema: string;
+    table: string;
+}
+
+const queryTableSchema = ({
+    database,
+    schema,
+    table,
+}: TableInfo) => `SELECT table_catalog
+                , table_schema
+                    , table_name
+                    , column_name
+                    , data_type
+            FROM ${database}.information_schema.columns
+            WHERE table_catalog = lower('${database}')
+                AND table_schema = lower('${schema}')
+                AND table_name = lower('${table}')
+            ORDER BY 1, 2, 3, ordinal_position`;
+
+const convertDataTypeToDimensionType = (
+    type: AthenaTypes | string,
+): DimensionType => {
+    const typeWithoutTimePrecision = type.replace(/\(\d\)/, '');
+    switch (typeWithoutTimePrecision) {
+        case AthenaTypes.BOOLEAN:
+            return DimensionType.BOOLEAN;
+        case AthenaTypes.TINYINT:
+            return DimensionType.NUMBER;
+        case AthenaTypes.SMALLINT:
+            return DimensionType.NUMBER;
+        case AthenaTypes.INTEGER:
+            return DimensionType.NUMBER;
+        case AthenaTypes.BIGINT:
+            return DimensionType.NUMBER;
+        case AthenaTypes.REAL:
+            return DimensionType.NUMBER;
+        case AthenaTypes.DOUBLE:
+            return DimensionType.NUMBER;
+        case AthenaTypes.DECIMAL:
+            return DimensionType.NUMBER;
+        case AthenaTypes.DATE:
+            return DimensionType.DATE;
+        case AthenaTypes.TIMESTAMP:
+            return DimensionType.TIMESTAMP;
+        case AthenaTypes.TIMESTAMP_TZ:
+            return DimensionType.TIMESTAMP;
+        default:
+            return DimensionType.STRING;
+    }
+};
+
+export class AthenaWarehouseClient extends WarehouseBaseClient<CreateAthenaCredentials> {
+    connectionOptions: Pick<
+        CreateAthenaCredentials,
+        'outputLocation' | 'workgroup' | 'extraBotoArgs' | 'database'
+    >;
+
+    client: AthenaClient;
+
+    constructor(credentials: CreateAthenaCredentials) {
+        super(credentials);
+        const credObj: { credentials?: any } = {};
+        if (credentials.awsAccessKeyId && credentials.awsSecretKey) {
+            credObj.credentials = {
+                accessKeyId: credentials.awsAccessKeyId,
+                secretAccessKey: credentials.awsSecretKey,
+            };
+        }
+
+        this.client = new AthenaClient({
+            region: credentials.awsRegion,
+            ...credObj,
+        });
+
+        this.connectionOptions = {
+            outputLocation: credentials.outputLocation,
+            workgroup: credentials.workgroup,
+            database: credentials.database,
+            extraBotoArgs: credentials.extraBotoArgs,
+        };
+    }
+
+    private async checkQueryExequtionStateAndGetData(
```

Comment:
just a thought, isn't there any JS library that takes care of the below? for example in python there is PyAthena, that pretty much take care of all the operations around firing a query and returning the result. I'm not really familiar with the JS world, but did you do a quick search to see if we can reusable some library? it will definitely reduce complexity.

Reply:
There's [athena-express](https://www.npmjs.com/package/athena-express) - i would prefer that, but didn't want to add too many dependencies. I'll take this as a green light and use it :)

Reply:
Perfect, just double check with the lightdash team and let's check if `[athena-express](https://www.npmjs.com/package/athena-express)` is well mantained. I see that it was published the last time 2 years ago, that is not a good indicator tbh.